appscale/tools/agents/azure_agent.py

```diff
@@ -829,16 +978,40 @@ def delete_vmss_instance(self, compute_client, parameters, vmss_name, instance_i
     """
     resource_group = parameters[self.PARAM_RESOURCE_GROUP]
     verbose = parameters[self.PARAM_VERBOSE]
-    AppScaleLogger.verbose("Deleting Virtual Machine Instance {0} from Scale "
-      "Set {1} ...".format(instance_id, vmss_name), verbose)
+
+    vm_info = "Virtual Machine {0} from Scale Set {1}".format(instance_id,
+                                                              vmss_name)
+
+    # Check if we succeeded deleting the instance before but timed out.
+    vm_list = compute_client.virtual_machine_scale_set_vms.list(
```

Comment:
Requiring the client to check first if the instance exists before issuing the delete call seems cumbersome. It's also possible for the instance to be created after the list and before the delete.

What happens if you issue the delete for an instance that doesn't exist?

Reply:
We waste about 6 minutes waiting for the instance to be deleted, if I remember correctly.

Reply:
I can double check on this though.

Reply:
there are no consequences to calling the delete without listing, so I will change this.